Makefile

```diff
@@ -57,17 +57,73 @@ svsm.bin.elf: $(OBJS) src/start/svsm.lds
 %.lds: %.lds.S src/start/svsm.h
 	$(GCC) $(A_FLAGS) $(LDS_FLAGS) -E -P -o $@ $<
 
-prereq: .prereq
+prereq: prereq_rust
+	touch .prereq
 
-.prereq:
```

Comment:
This can't be removed. If I do:
  rm .prereq
  make

Then I get "make: *** No rule to make target '.prereq', needed by 'all'.  Stop."

So you should probably change the all target to have a dependency on prereq_rust and then prereq_rust would have a dependency on .prereq. So the targets should look something like:

all: prereq_rust svsm.bin
doc: prereq_rust
prereq_rust: .prereq
.prereq

The touch of the .prereq file should be kept in the .prereq block.

Reply:
@tlendacky I'm wondering how we could let developers know that they need to re-rerun `make prereq` (or part of it) due to new features that have been added (e.g. openssl). 

I think a `prereq_check` target would be very helpful for that. We could still use the .prereq file as an evidence that rust has already been configured. However, we wouldn't need the `.prereq` target because it would be replaced by the `prereq_check` target.

It would be something like this without breaking the already established .prereq file purpose. Any thoughts?

```
all: prereq_check svsm.bin

prereq_check:
	@test -e .prereq || \
		{ echo "ERR: Please run 'make prereq' before proceeding"; exit 1; }
	@test -s external/openssl/Makefile || \
		{ echo "ERR: Please run 'make prereq_external' before proceeding"; exit 1; }

doc: prereq_check
	cargo doc --open

external: prereq_check
	$(MAKE) -C external/libcrt
	$(MAKE) -C external/openssl -j$$(nproc)

prereq: prereq_main prereq_external

prereq_main:
	curl --proto '=https' --tlsv1.2 -sSf https://sh.rustup.rs | sh
       <stripped prereq tasks>
	cargo install bootimage
	touch .prereq

prereq_external:
	git submodule update --init
	(cd external/openssl && git checkout OpenSSL_1_1_1q  && \
		./Configure \
			<stripped openssl parameters>
			--with-rand-seed=none \
			-I../libcrt/include \
			-Wl,rpath=../libcrt -lcrt )
```

Reply:
@tlendacky Hmm ... that's not good. I improved that part in the Makefile.